### PR TITLE
@arangodb: transaction execute can return any valid Javascript value.

### PR DESCRIPTION
**Pull request by jvenable-brex**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [_executeTransaction](<<https://www.arangodb.com/docs/stable/transactions-transaction-invocation.html>>): can return any legal JavaScript value from the function
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by pluma** [comment]
Would it make sense to make `Transaction` generic (`Transaction<ReturnType = any>`) and use the return type of the execute function instead of `any`?

**Comment by jvenable-brex** [comment]
I've templated the Transaction return type and retested in my code project; vscode is now showing me the proper return type from the invocation of db._executeTransaction and I'm still receiving the proper return value:

`const id = db._executeTransaction<string>({`

It's showing me the type of 'id' as being 'string'.

Thanks for the suggestion!

**Comment by jvenable-brex** [comment]
It appears that the master branch is broken from another merge:
`Error: ENOENT: no such file or directory, open './types/bell/index.d.ts'`

**Comment by davidalvarezr** [comment]
> It appears that the master branch is broken from another merge:
> `Error: ENOENT: no such file or directory, open './types/bell/index.d.ts'`

Same problem here #51144 and here #51150
Let’s open an issue ?

**Comment by davidalvarezr** [comment]
I opened an issue #51156

**Comment by jvenable-brex** [comment]
Everything's green.

**Review by pluma — APPROVED** [review]
lgtm

File: types/arangodb/index.d.ts [hunk 1]
```diff
@@ -907,7 +907,7 @@ declare namespace ArangoDB {
         write?: string | string[];
         allowImplicit?: boolean;
     }
-    interface Transaction {
+    interface Transaction<ReturnType = any> {
         collections: TransactionCollections | string[];
         action: (params: object) => void | string;
```
Comment (pluma) [inline types/arangodb/index.d.ts:912]: This should probably be `(params: object) => ReturnType` now?

**Review by pluma — CHANGES_REQUESTED** [review]

**Comment by pluma** [comment]
Could you please add a usage example to `arangodb-tests.ts`?

Maybe something like

```ts
const arr: string[] = db._executeTransaction({
  collections: ["test"],
  action() {
    return db._query("FOR d IN test RETURN d._key").toArray() as string[];
  }
});
```

This should satisfy the "new tests" requirement.

File: types/arangodb/index.d.ts (hunk 1 above)
Comment (jvenable-brex) [reply types/arangodb/index.d.ts:912]: Yup, that makes sense.

**Comment by jvenable-brex** [comment]
What's the procedure for executing the test? It looks like a foxx module given the router usage, but it's not exporting anything and there's no manifest.  Does it just run the script and look for errors?

**Comment by pluma** [comment]
@jvenable-brex I think "test" is a bit misnomer, it's more like a specimen. The DT tooling runs the test file through various versions of `tsc` using the provided definitions. It doesn't test for behavior, just for type correctness.

**Comment by jvenable-brex** [comment]
Ok, so the collections and database don't actually need to exist; understood!

File: types/arangodb/arangodb-tests.ts [hunk 2]
```diff
@@ -86,6 +86,27 @@ router.get("/", (req, res) => {
     }
 });
 
+router.put(
+    (request: Foxx.Request, response: Foxx.Response) => {
+        try {
+            const id = db._executeTransaction<string>({
```
[the remark this answers is not in the document]
Comment (pluma) [reply types/arangodb/arangodb-tests.ts:92]: To tell the test script that this should be inferred as a string the const should be prefixed with a type expectation comment like so:

```
// $ExpectType string
const id = db._executeTransaction({
...
```

See https://github.com/DefinitelyTyped/DefinitelyTyped/#my-package-teststs

**Review by pluma — CHANGES_REQUESTED** [review]

File: types/arangodb/arangodb-tests.ts [hunk 3]
```diff
@@ -86,6 +86,27 @@ router.get("/", (req, res) => {
     }
 });
 
+router.put(
```
[the remark this answers is not in the document]
Comment (jvenable-brex) [reply types/arangodb/arangodb-tests.ts:89]: Ahh, I was following the guidance on "usage example", so I felt it appropriate.

File: types/arangodb/arangodb-tests.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvenable-brex) [reply types/arangodb/arangodb-tests.ts:92]: Done.

File: types/arangodb/arangodb-tests.ts [hunk 4]
```diff
@@ -86,6 +86,27 @@ router.get("/", (req, res) => {
...
+    (request: Foxx.Request, response: Foxx.Response) => {
+        try {
+            const id = db._executeTransaction<string>({
+                collections: {
+                    read: "users",
+                    write: ["groups", "member"],
+                    allowImplicit: false,
+                },
+                action: (params: object): string => {
```
[the remark this answers is not in the document]
Comment (jvenable-brex) [reply types/arangodb/arangodb-tests.ts:98]: Done.

**Review by pluma — APPROVED** [review]
lgtm

**Comment by pluma** [comment]
ready to merge

**Pull request merged**